tutorials/identifying-first-day-no-snow/index.md

```diff
@@ -0,0 +1,516 @@
+day with zero percent snow cover. For applications in the northern
+hemisphere, you will likely want to start with January 1st (DOY `1`).
+However, if you are studying snowmelt timing in the southern hemisphere (e.g.,
+the Andes), where snowmelt can occur on dates either side of the new year,
+it is more appropriate to start the year on July 1st (DOY `183`), for
+instance. In this calculation, a year is defined as the 365 days beginning
+from the provided `startDay`.
+
+```js
+var startDoy = 1;
+```
+
+Define the year to start and end tracking snow cover fraction. All years in
+the range will be included in the analysis.
+
+```js
+var startYear = 2000;
+var endYear = 2019;
+```
+
+### 2\. Define date bands
+
+Define a function to add several date bands to the images; they will be used
+in a future step. Each image has a metadata timestamp property, but since we
+will be creating annual image mosaics composed of pixels from many different
+images, the date needs to be encoded per pixel as a value in an image band so
+that it is retained in the final mosaics. The function encodes:
+
+- Calendar DOY (`calDoy`): enumerated DOY from January 1st
+- Relative DOY (`relDoy`): enumerated DOY from user-defined `startDoy`
+- Milliseconds from Unix epoch (`millis`)
+- Year (`year`) - Note that the year is tied to the `startDoy` e.g., if
+`startDoy` is set at 183, the analysis will cross the new year, the year
+given to all pixels will be the first year, even if a particular image was
+collected on January 1 of the subsequent year.
+
+Note that two global variables are initialized (`startDate` and `startYear`)
+that will be redefined iteratively in a subsequent step.
+
+```js
+var startDate;
+var startYear;
+
+function addDateBands(img) {
+  // Get image date.
+  var date = img.date();
+  // Get calendar day-of-year.
+  var calDoy = date.getRelative('day', 'year');
+  // Get relative day-of-year; enumerate from user-defined startDoy.
+  var relDoy = date.difference(startDate, 'day');
+  // Get the date as milliseconds from Unix epoch.
+  var millis = date.millis();
+  // Add all of the above date info as bands to the snow fraction image.
+  var dateBands = ee.Image.constant([calDoy, relDoy, millis, startYear])
+    .rename(['calDoy', 'relDay', 'millis', 'year']);
+  return img.addBands(dateBands).toInt16().set('millis', millis);
+}
+```
+
+### 3\. Define an analysis mask
+
+Here is the opportunity to define a mask for your analysis. This mask can be
+used to constrain the analysis to certain latitudes
+(`ee.Image.pixelLonLat()`), land cover types, geometries, etc. In this case
+we will: 1) mask out water so that the analysis is confined to pixels over
+landforms only; 2) mask out pixels that have very few days of snow cover; 3)
+mask out pixels that are snow covered for a good deal of the year
+(e.g., glaciers).
+
+Import the MODIS water/land mask dataset, select the ‘water_mask’ band, and
+set all land pixels to value 1.
```

Comment:
Consider replacing final "." with ":" for this code snippet heading and the ones below.

Reply:
Done